pkg/util/instancemetadata/prod_test.go

```diff
@@ -115,16 +143,21 @@ func TestPopulateInstanceMetadata(t *testing.T) {
 				t.Error(err)
 			}
 
-			if p.subscriptionID != tt.wantSubscriptionID {
-				t.Error(p.subscriptionID)
-			}
-
-			if p.location != tt.wantLocation {
-				t.Error(p.location)
-			}
-
-			if p.resourceGroup != tt.wantResourceGroup {
-				t.Error(p.resourceGroup)
+			if err == nil {
```

Comment:
don't do that - change t.Error(err) above to t.Fatal(err) and have done with it.